### PR TITLE
Mention API naming consistency

### PR DESCRIPTION
**Pull request by milseman**

### CODE REVIEW

File: Documentation/Evolution/StringProcessingAlgorithms.md [hunk 1]
```diff
@@ -640,6 +640,46 @@ extension RangeReplaceableCollection where Element: Equatable {
...
+
+The protocol requirement is `var regex: Regex<RegexOutput>`, i.e. any type that can produce a regex or hook into the engine's customization hooks (this is what `consuming` does) can be used as a component of the DSL and with these generic API. An alternative name could be "CustomRegexConvertible", but we don't feel that communicates component composability very well, nor is it particularly enlightening when encountering these generic API.
+
+Another alternative is to have a second protocol just for generic API. But without a compelling semantic distinction or practical utility, we'd prefer to avoid adding protocols just for names. If a clearly superior name exists, we should just choose that.
+
+
+```
```
Comment (itingliu) [inline Documentation/Evolution/StringProcessingAlgorithms.md:669]:
```suggestion
```swift
```

File: Documentation/Evolution/StringProcessingAlgorithms.md [hunk 2]
```diff
@@ -640,6 +640,46 @@ extension RangeReplaceableCollection where Element: Equatable {
...
+### Cross-proposal API naming consistency
+
+The regex work is broken down into 6 proposals based on technical domain, which is advantageous for deeper technical discussions and makes reviewing the large body of work manageable. The disadvantage of this approach is that relatively-shallow cross-cutting concerns, such as API naming consistency, are harder to evaluate until we've built up intuition from multiple proposals.
+
+We've seen the [Regex type and overview](https://github.com/apple/swift-evolution/blob/main/proposals/0350-regex-type-overview.md), the [Regex builder DSL](https://github.com/apple/swift-evolution/blob/main/proposals/0351-regex-builder.md), and here we present lots of ways to use regex. Now's a good time to go over API naming consistency.
+
+(The other proposal with a significant amount of API is [Unicode for String Processing](https://forums.swift.org/t/pitch-unicode-for-string-processing/56907), which is in the pitch phase. It is a technical niche and less impactful on these naming discussions. We'll still want to design those names for consistency, of course.)
+
+
+```
```
Comment (itingliu) [inline Documentation/Evolution/StringProcessingAlgorithms.md:652]:
```suggestion
```swift
```

**Review by itingliu — APPROVED** [review]

File: Documentation/Evolution/StringProcessingAlgorithms.md [hunk 3]
```diff
@@ -640,6 +640,46 @@ extension RangeReplaceableCollection where Element: Equatable {
...
+
+The protocol requirement is `var regex: Regex<RegexOutput>`, i.e. any type that can produce a regex or hook into the engine's customization hooks (this is what `consuming` does) can be used as a component of the DSL and with these generic API. An alternative name could be "CustomRegexConvertible", but we don't feel that communicates component composability very well, nor is it particularly enlightening when encountering these generic API.
+
+Another alternative is to have a second protocol just for generic API. But without a compelling semantic distinction or practical utility, we'd prefer to avoid adding protocols just for names. If a clearly superior name exists, we should just choose that.
+
+
+```swift
+protocol CustomConsumingRegexComponent {
+    func consume(...)
```
Comment (rxwei) [inline Documentation/Evolution/StringProcessingAlgorithms.md:671]:
```suggestion
    func consuming(...)
```

**Comment by milseman** [comment]
@swift-ci please test

**Comment by milseman** [comment]
@swift-ci please test

**Pull request merged**